src/components/SelectionList/Search/ActionCell.tsx

```diff
@@ -41,15 +52,33 @@ function ActionCell({
     isChildListItem = false,
     parentAction = '',
     isLoading = false,
+    policyID = '',
+    reportID = '',
+    hash,
+    amount,
 }: ActionCellProps) {
     const {translate} = useLocalize();
     const theme = useTheme();
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
     const {isOffline} = useNetwork();
 
+    const [iouReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {canBeMissing: true});
     const text = isChildListItem ? translate(actionTranslationsMap[CONST.SEARCH.ACTION_TYPES.VIEW]) : translate(actionTranslationsMap[action]);
     const shouldUseViewAction = action === CONST.SEARCH.ACTION_TYPES.VIEW || (parentAction === CONST.SEARCH.ACTION_TYPES.PAID && action === CONST.SEARCH.ACTION_TYPES.PAID);
+    const [chatReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${iouReport?.chatReportID}`, {canBeMissing: true});
+    const bankAccountRoute = getBankAccountRoute(chatReport);
+
+    const confirmPayment = useCallback(
+        (type: PaymentMethodType | undefined) => {
+            if (!type || !reportID || !hash || !amount) {
+                return;
+            }
+
+            payMoneyRequestOnSearch(hash, [{amount, paymentType: type, reportID}]);
+        },
+        [hash, amount, reportID],
+    );
```

Comment:
Do you think we can reuse the logic of `handleActionButtonPress` instead of creating a seprate logic in here?